File: 1437-check-if-all-1s-are-at-least-length-k-places-away.java
class CheckIfAll1sAreAtLeastLengthKPlacesAway {
    public boolean kLengthApart(int[] nums, int k) {
        int prev = -k - 1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0){
                continue;
            }

            if (i - prev - 1 < k) {
                return false;
            }
            prev = i;
        }
        return true;
    }
  
    public static void main(String[] args) {
        CheckIfAll1sAreAtLeastLengthKPlacesAway sol = new CheckIfAll1sAreAtLeastLengthKPlacesAway();
        System.out.println("Output: " + sol.kLengthApart(new int[]{1, 0, 0, 0, 1, 0, 0, 1}, 2) + ", expected: True");
        System.out.println("Output: " + sol.kLengthApart(new int[]{1, 0, 0, 1, 0, 1}, 2) + ", expected: False");
    }
}

File: 1437-check-if-all-1s-are-at-least-length-k-places-away.py
class Solution:
    def kLengthApart(self, nums: list[int], k: int) -> bool:
        '''
        We go through each num, using a `prev` variable, if it is a `1`, and `current - prev - 1` isn't at least `k` return False
        '''
        prev = float('-inf')
        for i, num in enumerate(nums):
            if not num:
                continue

            if i - prev - 1 < k:
                return False
            prev = i

        return True


sol = Solution()
print(
    f'output: {sol.kLengthApart(nums=[1, 0, 0, 0, 1, 0, 0, 1], k=2)}, expected: True')
print(
    f'output: {sol.kLengthApart(nums=[1, 0, 0, 1, 0, 1], k=2)}, expected: False')
